README.md

```diff
@@ -20,9 +20,54 @@ Via Composer
 $ composer require php-http/httplug-bundle
 ```
 
+Enable the bundle in your kernel:
+ 
+``` php
+<?php
+// app/AppKernel.php
+
+public function registerBundles()
+{
+    $bundles = array(
+        // ...
+        new Http\HttplugBundle\HttplugBundle(),
+    );
+}
+```
 
 ## Usage
 
+The usage documentation is split into two parts. First we explain how to configure the bundle in an application. The second part is for developing reusable Symfony bundles that depend on an HTTP client defined by the Httplug interface.
+
+### Use in applications
+
+This bundle provides 3 services: 
+
+* `httplug.client` a service that provides the `Http\Client\HttpClient`
+* `httplug.message_factory` a service that provides the `Http\Message\MessageFactory`
+* `httplug.uri_factory` a service that provides the `Http\Message\UriFactory`
+
+These services are always an alias to another service. You can specify your own service or leave the default, which is the same name with `.default` appended. The default services in turn use the service discovery mechanism to provide the best available implementation. You can specify a class for each of the default services to use instead of discovery. 
+
+If you need to customize the service with decorators, e.g. to add authentication headers, decorate the default service and configure the main alias to your decorating service name.
+
+```yaml
+httplug:
+    main_alias:
+        client: httplug.client.default
+        message_factory: httplug.message_factory.default
+        uri_factory: httplug.uri_factory.default
+    classes:
+        client: ~ # uses discovery if not specified
```

Comment:
the user can specify a class for the client / adapter and we just
instantiate that class without arguments. in a first step, i would
expect the user to define a service for their adapter if they need
specific configuration on their client/adapter.

the next step will be to allow to configure plugins and wrapper clients.
i want to avoid configuring the actual guzzle *client*, but if the
guzzle *adapter* has specific configuration, we can add that at some
point. not sure how much specific configuration the adapters will have
however.